### PR TITLE
Patches For 2158

### PR DESCRIPTION
**Pull request by pavelbrm**
This PR provides further feedback for #2158:
- The checking functionality has been extracted into a helper.
    - This helps keep logic decoupled, and allows for testing in isolation (which results in better and easy to understand tests).
- The operations for obtaining numeric values and corresponding error messages for failure cases have been fixed and improved.
    - Obtaining of the numbers from order metadata contained an error – the code tried to type-cast, but the returned error indicated that a value was missing. If we are concerned with indicating whether a value is missing or not, we should split the obtaining and type-casting steps – and report accordingly. The joined type-cast-while-obtaining won't be able to disambiguate a missing value from the value of a wrong type.
- The operations for obtaining numeric values have been moved to where they belong – in the order context. Service must not know about how those values are represented.
- The helper does only one check per item's `CredentialType`, as it should. The originally proposed code checked the same order item twice – one `if` for `single-use`, and then another `if` _right away_ for `time-limited-v2`, although this does not make sense as those two cases are mutually exclusive.
- As a result, better tests – more focussed on what is being tested, without relying on side effects.
    - The originally proposed tests completely obfuscated what was being tested, relying on side effects and other places. The outcome – code (tests must be treated as production code) that is tightly coupled, hard to follow, maintain and change.
- Fixed other small inaccuracies (e.g. unnecessary global variables and/or cases where constants could be used instead).

### CODE REVIEW

File: services/skus/credentials.go [hunk 1]
```diff
@@ -673,3 +649,37 @@ func (s *Service) DeleteOrderCreds(ctx context.Context, orderID uuid.UUID, isSig
...
+			return err
+		}
+
+		if ncreds > nperInterval*nintervals {
+			return errInvalidNCredsTlv2
+		}
+
+		return nil
+	default:
+		return nil
```
Comment (clD11) [inline services/skus/credentials.go:683]: question: is it safe to return nil if we receive an unknown credential type?

Comment (pavelbrm) [reply services/skus/credentials.go:683]: Good question. This refactoring is based on the existing assumptions.

I would expect the caller to take care about that, and this code can be left for now. However, we can also return a sentinel error and handle it. But this would complicate handling of that error at the call site.

What do you think?

**Review by husobee — APPROVED** [review]
